### PR TITLE
Merge APIM Gateway and CI bits into main

### PR DESCRIPTION
**Pull request by nimakamoosi**
Merge latest changes used in APIM Gateway as well as CI bits that help us verify and validate the changes.

### CODE REVIEW

File: build/Dependencies.System2.props [hunk 1]
```diff
@@ -95,7 +95,7 @@
 
     <SystemTextEncodingCodePages2PackageVersion>4.5.0</SystemTextEncodingCodePages2PackageVersion>
     <SystemTextJson2PackageVersion>4.7.1</SystemTextJson2PackageVersion>
-    <SystemTextEncodingsWeb2PackageVersion>4.7.2</SystemTextEncodingsWeb2PackageVersion>
+    <SystemTextEncodingsWeb2PackageVersion>4.6.0</SystemTextEncodingsWeb2PackageVersion>
```
Comment (tomkerkhove) [inline build/Dependencies.System2.props]: Are we downgrading all of these on purpose?

File: build/templates/install-build-dependencies.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+parameters:
+  packageFeed: ''
+
+steps:
+# - task: NuGetCommand@2
```
Comment (tomkerkhove) [inline build/templates/install-build-dependencies.yaml:5]: Is this on purpose? This used to be required to build

Comment (nimakamoosi) [reply build/templates/install-build-dependencies.yaml:5]: Abandoned this PR in favor of staged PRs starting with [0] main -> [1]=+ CI -> [2] =+ APIMGateway -> [3] =+ Upstream -> [4] =+tlsAsyncWrite()
But will keep this in mind and ask you about correctness of future PRs

File: build/Dependencies.System2.props (hunk 1 above)
Comment (nimakamoosi) [reply build/Dependencies.System2.props]: Abandoned this PR in favor of staged PRs starting with [0] main -> [1]=+ CI -> [2] =+ APIMGateway -> [3] =+ Upstream -> [4] =+tlsAsyncWrite()
But will keep this in mind and ask you about correctness of future PRs

**Pull request closed without merging**